lineapy/graph_reader/artifact_collection.py

```diff
@@ -225,146 +206,57 @@ def _sort_session_artifacts(
-                )
-                for coll in session_artifacts.artifact_nodecollections
-            ]
-        )
-
-        # Generate session function return value string
-        session_function_return = ", ".join(
-            [
-                coll.return_variables[0]
-                for coll in session_artifacts.artifact_nodecollections
-                if coll.collection_type == NodeCollectionType.ARTIFACT
-            ]
-        )
-
-        session_input_parameters_body = session_artifacts.input_parameters_nodecollection.get_input_parameters_block(
-            indentation=indentation
-        )
-
-        SESSION_FUNCTION_TEMPLATE = load_plugin_template(
-            "session_function.jinja"
-        )
-        session_function = SESSION_FUNCTION_TEMPLATE.render(
-            session_input_parameters_body=session_input_parameters_body,
-            indentation_block=indentation_block,
-            session_function_name=session_function_name,
-            session_function_body=session_function_body,
-            return_dict_name=return_dict_name,
-        )
-
-        session_input_parameters = ", ".join(
-            session_artifacts.input_parameters_node.keys()
-        )
-
-        # Generate calculation code block for the session
-        # This is to be used in multi-session module
-        session_calculation = f"{indentation_block}artifacts.update({session_function_name}({session_input_parameters}))"
-
-        return {
-            "session_imports": session_imports,
-            "artifact_functions": artifact_functions,
-            "session_function": session_function,
-            "session_function_return": session_function_return,
-            "session_calculation": session_calculation,
-            "session_input_parameters_body": session_input_parameters_body,
-        }
-
     def _compose_module(
         self,
         session_artifacts_sorted: List[SessionArtifacts],
         indentation: int = 4,
+        return_dict_name="artifacts",
     ) -> str:
         """
         Generate a Python module that calculates artifacts
         in the given artifact collection.
         """
         indentation_block = " " * indentation
 
-        # Extract module script components by session
-        session_modules = [
-            self._extract_session_module(
-                session_artifacts=session_artifacts,
-                indentation=indentation,
-            )
-            for session_artifacts in session_artifacts_sorted
-        ]
-
-        # Combine components by "type"
         module_imports = "\n".join(
-            [module["session_imports"] for module in session_modules]
+            [
+                sa.get_session_module_imports()
+                for sa in session_artifacts_sorted
+            ]
         )
-        artifact_functions = "\n".join(
-            [module["artifact_functions"] for module in session_modules]
+
+        artifact_function_definition = "\n".join(
```

Comment:
NIT: For clarity, I suggest using plural for the variable name, i.e., `artifact_function_definitions`